src/purview.ts

```diff
@@ -65,8 +65,9 @@ interface WebSocketStateNoCSS extends BaseWebSocketState {
 interface BaseWebSocketState {
   ws: WebSocket
   roots: ConnectedRoot[]
-  connected: boolean
+  connectionState: null | "connecting" | "connected"
```

Comment:
I chose `null` as the third possible value to represent that nothing has been done for the connection state. I'm cool with changing it to something else like "idle", "waiting", or alike, but I felt `null` was most representative